### PR TITLE
[Model] add model PNA

### PR DESCRIPTION
**Pull request by joker666666666**
[Doc] add file readme.md. 
describe error happen in paddle backend.

### CODE REVIEW

File: examples/pna/pna_trainer.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
+# !/usr/bin/env python
+# -*- encoding: utf-8 -*-
+"""
+@File    :   pna_trainer.py
+@Time    :   2022/5/26 20:07:55
+@Author  :   huang le
+"""
+import os
+os.environ['TL_BACKEND'] = 'paddle'
```
[the remark this answers is not in the document]
Comment (joker666666666) [reply examples/pna/pna_trainer.py:9]: ok！

File: examples/pna/pna_trainer.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+from tensorlayerx import convert_to_tensor, convert_to_numpy
+from gammagl.datasets import ZINC
+from gammagl.loader import DataLoader
+from gammagl.models import PNAModel
+from gammagl.utils.degree import degree
+from tensorlayerx.model import TrainOneStep, WithLoss
+from tensorlayerx.losses import absolute_difference_error
+from tensorlayerx.optimizers.lr import ReduceOnPlateau
+
+tlx.set_device(device='GPU', id=0)
```
[the remark this answers is not in the document]
Comment (joker666666666) [reply examples/pna/pna_trainer.py:26]: ok!

File: gammagl/layers/conv/pna_conv.py [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+    def forward(self, x, edge_index, edge_attr=None):
+        if self.divide_input:
+            x = tlx.reshape(x, (-1, self.towers, self.F_in))
+        else:
+            x = tlx.stack([x for i in range(0, self.towers)], axis=1)
+
+        out = self.propagate(x=x, edge_index=edge_index, edge_attr=edge_attr)
+        out = tlx.concat([x, out], axis=-1)
+        outs = [nn(out[:, i]) for i, nn in enumerate(self.post_nns)]
+        out = tlx.concat(outs, axis=1)
```
[the remark this answers is not in the document]
Comment (joker666666666) [reply gammagl/layers/conv/pna_conv.py:132]: Yes. The variables out and outs are both tensors. The last two lines use Multiple Towers in order to parallelize the processing and speed up the training.

File: gammagl/layers/conv/pna_conv.py [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+
+        if edge_attr is not None:
+            edge_attr = self.edge_encoder(edge_attr)
+            edge_attr = tlx.stack([edge_attr for i in range(0, self.towers)], axis=1)
+            h = tlx.concat([x_i, x_j, edge_attr], axis=-1)
+        else:
+            h = tlx.concat([x_i, x_j], axis=-1)
+
+        hs = [nn(h[:, i]) for i, nn in enumerate(self.pre_nns)]
+        return tlx.stack(hs, axis=1)
```
[the remark this answers is not in the document]
Comment (joker666666666) [reply gammagl/layers/conv/pna_conv.py:147]: Ditto.

File: gammagl/layers/conv/pna_conv.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (dddg617) [reply gammagl/layers/conv/pna_conv.py:132]: Right, nice work.

File: gammagl/models/pna.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        self.node_emb = nn.Embedding(21, 75)
+        self.edge_emb = nn.Embedding(4, 50)
+        self.convs = nn.ModuleList()
+        self.batch_norms = nn.ModuleList()
+
+        aggregators = aggregators.split()
+        scalers = scalers.split()
+
+        for _ in range(4):
+            conv = PNAConv(in_channels=75, out_channels=75,
```
[the remark this answers is not in the document]
Comment (joker666666666) [reply gammagl/models/pna.py:21]: Ok. I will fix that.

File: gammagl/layers/conv/pna_conv.py [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+            for _ in range(pre_layers - 1):
+                modules += [ReLU()]
+                modules += [Linear(in_features=self.F_in, out_features=self.F_in)]
+            self.pre_nns.append(Sequential(modules))
+
+            in_channels = (len(aggregators) * len(scalers) + 1) * self.F_in
+            modules = [Linear(in_features=in_channels, out_features=self.F_out)]
+            for _ in range(post_layers - 1):
+                modules += [ReLU()]
+                modules += [Linear(in_features=self.F_out, out_features=self.F_out)]
```
[the remark this answers is not in the document]
Comment (joker666666666) [reply gammagl/layers/conv/pna_conv.py:118]: I will try to use nn.Sequential.

File: gammagl/layers/pool/glob.py [hunk 7]
```diff
@@ -80,4 +80,31 @@ def global_max_pool(x, batch, size: Optional[int] = None):
...
+    return tlx.unsorted_segment_max(x, batch, size)
+
+
+def global_min_pool(x, batch, size: Optional[int] = None):
+    r"""Returns batch-wise graph-level-outputs by taking the channel-wise
+    maximum across the node dimension, so that for a single graph
+    :math:`\mathcal{G}_i` its output is computed by
+
+    .. math::
+        \mathbf{r}_i = \mathrm{max}_{n=1}^{N_i} \, \mathbf{x}_n
```
[the remark this answers is not in the document]
Comment (joker666666666) [reply gammagl/layers/pool/glob.py]: Completed.

File: gammagl/layers/conv/pna_conv.py [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+            there are any). (default :obj:`None`)
+        towers (int, optional): Number of towers (default: :obj:`1`).
+        pre_layers (int, optional): Number of transformation layers before
+            aggregation (default: :obj:`1`).
+        post_layers (int, optional): Number of transformation layers after
+            aggregation (default: :obj:`1`).
+        divide_input (bool, optional): Whether the input features should
+            be split between towers or not (default: :obj:`False`).
+        **kwargs (optional): Additional arguments of
+            :class:`torch_geometric.nn.conv.MessagePassing`.
```
Comment (dddg617) [inline gammagl/layers/conv/pna_conv.py:64]: Change ``torch_geometric`` into ``tlx`` or delete this.

File: gammagl/layers/conv/pna_conv.py [hunk 9]
```diff
@@ -0,0 +1,199 @@
+import tensorlayerx as tlx
+import tensorlayerx.nn
+from tensorlayerx import reduce_sum
+from tensorlayerx.nn import ReLU, Linear, ModuleList, Sequential
+# from torch_scatter import scatter
```
Comment (dddg617) [inline gammagl/layers/conv/pna_conv.py:5]: Delete this line.

**Pull request merged**